### PR TITLE
Limit the branches that travis builds on

### PR DESCRIPTION
**Pull request by carols10cents**
Because branches that become PRs are built twice :(

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,12 @@ language: rust
 sudo: required
 dist: xenial
 
+branches:
+  only:
+  - auto
+  - try
```
Comment (smarnach) [inline .travis.yml]: What is this branch? We don't currently seem to have a branch with this name.

Comment (pietroalbini) [reply .travis.yml]: It's used when you do `@bors try`.

**Comment by jtgeibel** [comment]
I saw it mentioned somewhere that we don't need to run tests on `master`, so I wanted to mention that we do want to continue building on master because Travis uses the "default branch" ([docs](https://docs.travis-ci.com/user/caching/#fetching-and-storing-caches)) as the fallback cache for new branches and PRs.  I don't see any UI for this in the Travis settings, and I don't think it makes sense to switch `HEAD` to point to `auto` instead of `master` on GitHub.

@bors r+

**Comment by smarnach** [comment]
Have we reached some quota limit for Travis builds? This didn't even start building on Travis yet, same for #1929.

**Comment by Turbo87** [comment]
@smarnach I'm not sure if it's on the TravisCI side since the PR jobs for Travis seem to run fine. does @bors have some sort of quota limit? 🤔

**Comment by Turbo87** [comment]
hmm, or @bors can run X number of jobs in parallel and the jobs that "timed out" are still running and blocking new jobs from being accepted 🤔

**Pull request merged**